subsys/bluetooth/controller/ll_sw/pdu.h

```diff
@@ -873,7 +873,13 @@ struct pdu_iso {
 	uint8_t hdr_other:6;
 	uint8_t ll_id:2;
 #endif /* CONFIG_LITTLE_ENDIAN */
-	uint8_t length;
+
+	uint8_t len;
+
+	union {
+		uint8_t resv;
+	};
```

Comment:
This is a non-standard field that is specific to the Nordic radio and should not be unconditionally part of the definition of a PDU. (It is bad enough that it is part of some existing definitions, but we should at least not make the problem worse.)

Reply:
Would it be acceptable to have a `pdu_vendor.h` with:

```c
/* Presence of vendor PDU struct octet3 */
struct pdu_data_vnd_octet3 {
	union {
		uint8_t resv;
	};
} __packed;
```

or

```c
/* No vendor PDU struct octet3*/
struct pdu_data_vnd_octet3 {
	uint8_t resv[0];
} __packed;
```

and include:

```c
	struct pdu_data_vnd_octet3 vnd_octet3;
```

Reply:
Having a `pdu_vendor.h` sounds fine to me. However, putting a zero-length array in the middle of another struct won't fly due to the C standard. Maybe a define like `PDU_DATA_VENDOR_EXT_HEADER` could be an option?

Reply:
GCC built fine when I have a zero-length array in between in the pdu_data struct, we have zero-length array at end of struct allover HCI too.

Reply:
> Having a `pdu_vendor.h` sounds fine to me. However, putting a zero-length array in the middle of another struct won't fly due to the C standard. Maybe a define like `PDU_DATA_VENDOR_EXT_HEADER` could be an option?

How so? Is this actually forbidden somewhere?

Reply:
@wopu-ot With the direction finding extensions, the 3rd Octet (optional) is a BT Standard for data channel PDUs. Should we just agree to keep the existing union? Is it an issue for your implementation (the octet is already upstream)?

Reply:
@carlescufi I am afraid I confused the rules for flexible-length arrays and zero-length arrays. Zero-length arrays in the middle of things are not a problem from the standard's perspective as far as I understand now. Sorry for the confusion.

Reply:
@cvinayak I would be happy to get rid of the additional byte, but we can live with what is upstream now. I would like to avoid making the issue worse by adding the `resv` field in more places though.

Reply:
@wopu-ot  Is pdu_vendor.h an acceptable solution, vendors can omit the octet3. See, https://github.com/zephyrproject-rtos/zephyr/pull/50926/commits/749e67597c2cd1230588af6f3b7fd6e808d3bacb#diff-8d21a201cb026a71cf58c6f34d9ee48816ff8ec493cb6426d02be884bc498b8eR1

Reply:
I believe the pdu_vendor.h solution should work for our end.